app/lnd/lib/lightning.js

```diff
@@ -21,37 +26,111 @@ process.env.GRPC_SSL_CIPHER_SUITES =
-      lnd.getInfo({}, { deadline: getDeadline(2) }, err => {
-        if (err) {
-          return reject(err)
-        }
-        return resolve(lnd)
+class Lightning {
+  constructor() {
+    this.mainWindow = null
+    this.lnd = null
+    this.subscriptions = {
+      channelGraph: null,
+      invoices: null,
+      transactions: null
+    }
+  }
+
+  /**
+   * Connect to the gRPC interface and verify it is functional.
+   * @return {Promise<rpc.lnrpc.Lightning>}
+   */
+  async connect() {
+    const lndConfig = config.lnd()
+    const { host, rpcProtoPath, cert, macaroon } = lndConfig
+
+    // Verify that the host is valid before creating a gRPC client that is connected to it.
+    return await validateHost(host).then(async () => {
+      // Load the gRPC proto file.
+      const rpc = grpc.load(rpcProtoPath)
+
+      // Create ssl and macaroon credentials to use with the gRPC client.
+      const [sslCreds, macaroonCreds] = await Promise.all([
+        createSslCreds(cert),
+        createMacaroonCreds(macaroon)
+      ])
+      const credentials = grpc.credentials.combineChannelCredentials(sslCreds, macaroonCreds)
+
+      // Create a new gRPC client instance.
+      const lnd = new rpc.lnrpc.Lightning(host, credentials)
+
+      // Call the getInfo method to ensure that we can make successful calls to the gRPC interface.
+      return new Promise((resolve, reject) => {
+        lnd.getInfo({}, { deadline: getDeadline(2) }, err => {
+          if (err) {
+            return reject(err)
+          }
+          this.lnd = lnd
+          return resolve(lnd)
+        })
       })
     })
-  })
+  }
+
+  /**
+   * Discomnnect the gRPC service.
+   */
+  disconnect() {
+    this.unsubscribe()
+    this.lnd.close()
+  }
+
+  /**
+   * Hook up lnd restful methods.
+   */
+  lndMethods(event, msg, data) {
+    return methods(this.lnd, mainLog, event, msg, data)
+  }
+
+  /**
+   * Subscribe to all bi-directional streams.
+   */
+  subscribe(mainWindow) {
+    this.mainWindow = mainWindow
+    this.subscribeToChannelGraph()
+    this.subscribeToInvoices()
+    this.subscribeToTransactions()
+  }
+
+  /**
+   * Unsubscribe from all bi-directional streams.
+   */
+  unsubscribe() {
+    Object.keys(this.subscriptions).forEach(subscription => {
+      if (this.subscriptions[subscription]) {
+        this.subscriptions[subscription].cancel()
+      }
+    })
+  }
+
+  /**
+   * Subscribe to the channel graph stream.
+   */
+  subscribeToChannelGraph() {
+    this.subscriptions.channelGraph = subscribeToChannelGraph(this.mainWindow, this.lnd, mainLog)
+  }
+
+  /**
+   * Subscribe to the invoice stream.
+   */
+  subscribeToInvoices() {
+    this.subscriptions.invoices = subscribeToInvoices(this.mainWindow, this.lnd, mainLog)
+  }
+
+  /**
+   * Subscribe to the transaction stream.
+   */
+  subscribeToTransactions() {
+    this.subscriptions.transactions = subscribeToTransactions(this.mainWindow, this.lnd, mainLog)
+  }
```

Comment:
How about inlining these into `subscribe` to minimize the interface. `subscriptions` could then be an array produced by a map.

Reply:
I'm not quite sure what you mean about making subscriptions an array produced by a map, but I have refactored this to reduce the interface so that there is only the one `subscribe` method